CMakeLists.txt

```diff
@@ -152,15 +151,31 @@ if (BUILD_FORTRAN_WRAPPER)
 
     if (Fortran_COMPILER_NAME STREQUAL "gfortran")
         # gfortran
-        set (CMAKE_Fortran_FLAGS         "-fdefault-real-8")
-        set (CMAKE_Fortran_FLAGS_RELEASE "-O3")
-        set (CMAKE_Fortran_FLAGS_DEBUG   "-g")
+        set (CMAKE_Fortran_FLAGS "-fdefault-real-8" CACHE STRING
+             "Flags used by the Fortran compiler during all builds."
+             FORCE )
+        set (CMAKE_Fortran_FLAGS_RELEASE "-O3" CACHE STRING
+             "Flags used by the Fortran compiler during Release builds."
+             FORCE )
+        set (CMAKE_Fortran_FLAGS_DEBUG "-g" CACHE STRING
+             "Flags used by the Fortran compiler during Debug builds."
+             FORCE )
```

Comment:
Ok, thanks for your contribution. Keeping the discussion here: 
> Also, it's worth pointing out that the CMAKE_CXX_FLAGS_Profile are force cached in that same file and they function as expected. I duplicated this behavior in my commit.

Ok, I think we shouldn't do that. Could you remove it also in the other places where they are cached?

Reply:
I think you do want/need them cached. The `CMake_<LANG>_Flags*` already exist in the cache because they are defined by CMake when you enable the language `<LANG>`. You want to override the existing flags with the new compiler options, hence the `FORCE`; otherwise the defaults remain. And because these flags are already cached variables, you want the `CACHE`. I don't think you need the `CACHE` if you're strictly using the command line version of CMake; but if you're using the GUI, these variables *must* have the `CACHE` attribute. I do use the GUI, hence how I encountered this issue.

Reply:
Ok, I just tried (moving the `add_subdirectory`) as you did because I reproduced the install directory bug), and this is what I get without CACHE. 

`cmake -DBUILD_FORTRAN_WRAPPER=1 ..`

```
cd /home/rdb/git/Mutationpp/build/interface/fortran && /usr/sbin/gfortran -Dmutation___fortran_EXPORTS -I/home/rdb/git/Mutationpp/src/general -I/home/rdb/git/Mutationpp/src/kinetics -I/home/rdb/git/Mutationpp/src/numerics -I/home/rdb/git/Mutationpp/src/thermo -I/home/rdb/git/Mutationpp/src/transfer -I/home/rdb/git/Mutationpp/src/transport -I/home/rdb/git/Mutationpp/src/utilities -I/home/rdb/git/Mutationpp/src/gsi -I/usr/include/eigen3 -fdefault-real-8 -O3 -J. -fPIC -c /home/rdb/git/Mutationpp/interface/fortran/cwrapper_interface.f90 -o CMakeFiles/mutation++_fortran.dir/cwrapper_interface.f90.o
```
cmake -DBUILD_FORTRAN_WRAPPER=1 -DCMAKE_BUILD_TYPE=Debug ..

```
cd /home/rdb/git/Mutationpp/build/interface/fortran && /usr/sbin/gfortran -Dmutation___fortran_EXPORTS -I/home/rdb/git/Mutationpp/src/general -I/home/rdb/git/Mutationpp/src/kinetics -I/home/rdb/git/Mutationpp/src/numerics -I/home/rdb/git/Mutationpp/src/thermo -I/home/rdb/git/Mutationpp/src/transfer -I/home/rdb/git/Mutationpp/src/transport -I/home/rdb/git/Mutationpp/src/utilities -I/home/rdb/git/Mutationpp/src/gsi -I/usr/include/eigen3 -fdefault-real-8 -g -J. -fPIC -c /home/rdb/git/Mutationpp/interface/fortran/cwrapper_interface.f90 -o CMakeFiles/mutation++_fortran.dir/cwrapper_interface.f90.o
cd /home/rdb/git/Mutationpp/build/interface/fortran && /usr/sbin/c++ -Dmutation___fortran_EXPORTS -I/home/rdb/git/Mutationpp/src/general -I/home/rdb/git/Mutationpp/src/kinetics -I/home/rdb/git/Mutationpp/src/numerics -I/home/rdb/git/Mutationpp/src/thermo -I/home/rdb/git/Mutationpp/src/transfer -I/home/rdb/git/Mutationpp/src/transport -I/home/rdb/git/Mutationpp/src/utilities -I/home/rdb/git/Mutationpp/src/gsi -isystem /usr/include/eigen3 -g -g -fPIC -std=gnu++14 -MD -MT interface/fortran/CMakeFiles/mutation++_fortran.dir/cwrapper.cpp.o -MF CMakeFiles/mutation++_fortran.dir/cwrapper.cpp.o.d -o CMakeFiles/mutation++_fortran.dir/cwrapper.cpp.o -c /home/rdb/git/Mutationpp/interface/fortran/cwrapper.cpp
```

So for me flags are being passed correctly without CACHE. Can you please try from a clean `build` directory? Because I rarely saw compiler flags force cached. Also I don't see why things should be different between CLI / GUI. 

PS: The `-g` flag is passed twice.

Reply:
Okay, this is somewhat confusing... I am able to repeat what you show above when I drop my second commit and use the CMake CLI -- the Fortran flags are set properly. When I use the GUI, the values displayed in the GUI for the compiler flags are incorrect; they are the default (cached) variables, because the GUI relies on the cache.

![image](https://user-images.githubusercontent.com/84190711/147790353-5de43629-3528-4c54-80de-18fc1540a9bf.png)

Note though that the `CMAKE_CXX_FLAGS_PROFILE` are set to the updated values from [here](https://github.com/mutationpp/Mutationpp/blob/c9a924fde43216e0777a918dc5042eaa18ae98b5/CMakeLists.txt#L66), because they are cached.

 However, when I actually look at the flags the compiler is using in `interface/fortran/CMakeFiles/mutation++_fortran.dir/flags.make`, **_they are set properly_**.

![image](https://user-images.githubusercontent.com/84190711/147790308-a50b3bef-cdc3-41b6-b13e-00a342fb26b3.png)

After reading more about the [GUI and the cache](https://cmake.org/cmake/help/latest/guide/user-interaction/index.html#id13), it seems the GUI is best viewed as an editor for the cache.
I'll let you decide how you want to proceed. I don't think there is anything wrong with caching the variables, especially since cached versions of them exist, and I think it would be less confusing to anyone using the GUI.

Reply:
Yeah, that's why I hate GUIs. They very often make things more confusing. :)

Anyway, if you tell me that force caching those flags shows them correctly in the GUI as you would expect, I can be good with that! In that case be sure that all the flags are correctly cached both for Fortran and C++

Reply:
Anyway, I think you need to run the configure step twice to get the right values in the GUI. It's how it works. The first time it shows the default because the local cache did not get generated.

Reply:
If also force caching does not show correct values in the GUI, then let's not cache them. :)